mlflow/langchain/langchain_tracer.py

```diff
@@ -266,6 +269,23 @@ def on_llm_start(
         mlflow_messages = [ChatMessage(role="user", content=prompt) for prompt in prompts]
         set_span_chat_messages(span, mlflow_messages)
 
+        if tools := self._extract_tool_definitions(kwargs):
+            set_span_chat_tools(span, tools)
+
+    def _extract_tool_definitions(self, kwargs: dict[str, Any]) -> list[ChatTool]:
+        raw_tools = kwargs.get("invocation_params", {}).get("tools", [])
+        tools = []
+        for raw_tool in raw_tools:
+            # First, try to parse the raw tool dictionary as OpenAI-style tool
+            try:
+                tool = ChatTool.validate_compat(raw_tool)
+                tools.append(tool)
+            except MlflowException:
+                # If not OpenAI style, just try to extract the name and descriptions.
+                tool = ChatTool(name=raw_tool.get("name"), description=raw_tool.get("description"))
```

Comment:
ChatTool only has type and function fields?

Reply:
ahhh nice catch, lemme add a test for this pass